code/modules/spacepods/pod_fab.dm

```diff
@@ -0,0 +1,488 @@
+						<a href='?src=\ref[src];process_queue=1'>Try again</a> | <a href='?src=\ref[src];clear_temp=1'>Return</a><a>"}
+			return FALSE
+		remove_from_queue(1)
+		build_part(D)
+		D = listgetindex(queue, 1)
+	say("Queue processing finished successfully.")
+
+/obj/machinery/spod_part_fabricator/proc/list_queue()
+	var/list/output = list("<B>Queue contains:</b>")
+	if(!istype(queue) || !queue.len)
+		output += "<br>Nothing"
+	else
+		output += "<ol>"
+		var/i = 0
+		for(var/datum/design/D in queue)
+			i++
+			var/obj/part = D.build_path
+			output += "<li[!check_resources(D)?" style='color: #f00;'":null]>"
+			output += initial(part.name) + " - "
+			output += "[i>1?"<a href='?src=\ref[src];queue_move=-1;index=[i]' class='arrow'>&uarr;</a>":null] "
+			output += "[i<queue.len?"<a href='?src=\ref[src];queue_move=+1;index=[i]' class='arrow'>&darr;</a>":null] "
+			output += "<a href='?src=\ref[src];remove_from_queue=[i]'>Remove</a></li>"
+
+		output += "</ol>"
+		output += "\[<a href='?src=\ref[src];process_queue=1'>Process queue</a> | <a href='?src=\ref[src];clear_queue=1'>Clear queue</a>\]"
+	return jointext(output, "")
+
+/obj/machinery/spod_part_fabricator/proc/sync()
+	temp = "Updating local R&D database..."
+	updateUsrDialog()
+	sleep(30) //only sleep if called by user
+
+	for(var/obj/machinery/computer/rdconsole/RDC in oview(7,src))
+		if(!RDC.sync)
+			continue
+		for(var/v in RDC.files.known_tech)
+			var/datum/tech/T = RDC.files.known_tech[v]
+			files.AddTech2Known(T)
+		for(var/v in RDC.files.known_designs)
+			var/datum/design/D = RDC.files.known_designs[v]
+			files.AddDesign2Known(D)
+		files.RefreshResearch()
+		temp = "Processed equipment designs.<br>"
+		//check if the tech coefficients have changed
+		temp += "<a href='?src=\ref[src];clear_temp=1'>Return</a>"
+
+		updateUsrDialog()
+		say("Successfully synchronized with R&D server.")
+		return
+
+	temp = "Unable to connect to local R&D Database.<br>Please check your connections and try again.<br><a href='?src=\ref[src];clear_temp=1'>Return</a>"
+	updateUsrDialog()
+	return
+
+/obj/machinery/spod_part_fabricator/proc/get_resource_cost_w_coeff(datum/design/D, resource, roundto = 1)
+	return round(D.materials[resource]*component_coeff, roundto)
+
+/obj/machinery/spod_part_fabricator/proc/get_construction_time_w_coeff(datum/design/D, roundto = 1) //aran
+	return round(initial(D.construction_time)*time_coeff, roundto)
+
+/obj/machinery/spod_part_fabricator/attack_hand(mob/user)
+	if(!(..()))
+		return interact(user)
+
+/obj/machinery/spod_part_fabricator/interact(mob/user as mob)
+	var/dat, left_part
+	if (..())
+		return
+	user.set_machine(src)
+	var/turf/exit = get_step(src,(dir))
+	if(exit.density)
+		say("Error! Part outlet is obstructed.")
+		return
+	if(temp)
+		left_part = temp
+	else if(being_built)
+		var/obj/I = being_built.build_path
+		left_part = {"<TT>Building [initial(I.name)].<BR>
+							Please wait until completion...</TT>"}
+	else
+		switch(screen)
+			if("main")
```

Comment:
im hoping screens can be defines like SPOD_SCREEN_MAIN or something so i don't have to get a fucking aneurysm like i did when i refactored r&d consoles in my pr but i know this won't be nearly as bad but still.